### DISCUSSION TITLE
Type-safety for returned type of not-null filter on relations

### DISCUSSION THREAD

**Discussion by DamianGlowala** [prisma/prisma#12333] [Q&A] [answered]
Given the following code (with `"strickNullChecks": true`):
```
const sampleUser = await this.prisma.userAccount.findFirst({
    where: {
        userRelatedTableName: {
            isNot: null
        }
    },
    select: {
        userRelatedTableName: true
    }
});
```
... the return type is:
```
const sampleUser: {
    userRelatedTableName: UserRelatedTableName | null;
} | null
```

Is it feasible for Prisma to achieve the following, not-null type based on the `where: { (...) }` query?
```
const sampleUser: {
    userRelatedTableName: UserRelatedTableName;
} | null
```

**Comment by austincrim** [comment, marked as answer]
Hey @DamianGlowala 👋 ,

I don't think this is possible today. 

Could you open a [feature request](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=) for this? Our engineering teams will pick it up from there.

**Reply by DamianGlowala** [reply]
Thank you for the answer, I have opened a feature request https://github.com/prisma/prisma/issues/12366.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
